include/mata/alphabet.hh

```diff
@@ -0,0 +1,448 @@
+/* alphabet.hh -- File containing alphabets for automata
+ *
+ * This file is a part of libmata.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#ifndef _MATA_ALPHABET_HH
+#define _MATA_ALPHABET_HH
+
+#include <forward_list>
+#include <string>
+
+#include <mata/util.hh>
+
+namespace Mata {
+    namespace Nfa {
+        struct Nfa;
+    }
+}
+
+namespace Mata {
+        using Symbol = unsigned long;
+        using StringToSymbolMap = std::unordered_map<std::string, Symbol>;
+
+        template<typename T> using Sequence = std::vector<T>; ///< A sequence of elements.
+        using AutSequence = Sequence<Mata::Nfa::Nfa>; ///< A sequence of non-deterministic finite automata.
+        template<typename T> using PtrSequence = Sequence<T*>; ///< A sequence of pointers to elements.
+        using AutPtrSequence = PtrSequence<Mata::Nfa::Nfa>; ///< A sequence of pointers to non-deterministic finite automata.
+        using ConstAutPtrSequence = PtrSequence<const Mata::Nfa::Nfa>; ///< A sequence of pointers to const non-deterministic finite automata.
+
+        template<typename T> using RefSequence = Sequence<std::reference_wrapper<T>>; ///< A sequence of references to elements.
+        using AutRefSequence = RefSequence<Mata::Nfa::Nfa>; ///< A sequence of references to non-deterministic finite automata.
+        using ConstAutRefSequence = RefSequence<const Mata::Nfa::Nfa>; ///< A sequence of const references to non-deterministic finite automat
```

Comment:
We have to discuss what where we want to introduce new types such as `State`, `Symbol`, etc. `Symbol` in alphabet header file? Keep states specific for each automaton type? `Nfa::State`, `Afa::State`, etc.? Is it necessary? Will there be a use case for this?

This is a topic for discussion on Thursday.

Reply:
Solved. I kept there only alphabet related data types.